### ISSUE TITLE
linux "SetEvent" always return 0, won't it fail?

### ISSUE THREAD

**Issue by inaryart** [neosmart/pevents#22]
when `WaitForEvent` is not in a waiting state, using `setevent` should return failure

**Comment by mqudsi** [comment]
I’m sorry, can you clarify?

Example code that behaves differently on Linux and Windows would be helpful.

**Comment by inaryart** [comment]
```c++

neosmart::while(WaitForEvent(event,-1))){

	// now dong
	do_something();
}

```

- when the `thread1` is executing do_something, it is not in a waiting state(NOT CALLING WaitForEvent), SetEvent should return an error.
```C++

neosmart::SetEvent(event);

```
- linux: always return 0
https://github.com/neosmart/pevents/blob/1209b1fd1bd2e75daab4380cf43d280b90b45366/src/pevents.cpp#L363

- windows: an error will be returned when the notification fails
https://github.com/neosmart/pevents/blob/1209b1fd1bd2e75daab4380cf43d280b90b45366/src/pevents.cpp#L603

**Comment by yinsidij** [comment]
@inaryart I tried to repo what you described for SetEvent. However, seems like I didn't see the error. SetEvent for the same event twice would just return 1
Here is my example code
https://github.com/yinsidij/pevents/blob/master/win32/HelloWorld.cpp

Here is the output running in windows (ret = 1 twice, which indicates success)

Thread 14076 waiting for write event...
Thread 18516 sets event...
ret = 1, error = 0
WaitForSingleObject returns 0
Thread 18516 sets event...
ret = 1, error = 0

The output is the same no matter the event is auto-reset or manual-reset or initialState signaled or nonsignaled.
Am I missing anything?

**Comment by inaryart** [comment]
> @inaryart I tried to repo what you described for SetEvent. However, seems like I didn't see the error. SetEvent for the same event twice would just return 1
> Here is my example code
> https://github.com/yinsidij/pevents/blob/master/win32/HelloWorld.cpp
> 
> Here is the output running in windows (ret = 1 twice, which indicates success)
> 
> Thread 14076 waiting for write event...
> Thread 18516 sets event...
> ret = 1, error = 0
> WaitForSingleObject returns 0
> Thread 18516 sets event...
> ret = 1, error = 0
> 
> The output is the same no matter the event is auto-reset or manual-reset or initialState signaled or nonsignaled.
> Am I missing anything?

you are right, `setEvent` will be return `1`, so i think linux should also return the same value(`1`), so that I can judge whether the wake-up(`setevent`) operation failed

**Comment by yinsidij** [comment]
I think neosmart::event uses return value of 0 as success, which is oppose to windows convention (return value of 1 as success). So you might want to update up-level checker.

**Comment by inaryart** [comment]
I think this is normal code,  it should be part of the project. 
```c++

if(setevent(event) == 1){
   // then i known 
    failedCount++;
}
```
just as my suggestion...
